spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DiscoveryClientServiceInstanceListSupplier.java

```diff
@@ -53,7 +64,7 @@ public String getServiceId() {
 
 	@Override
 	public Flux<List<ServiceInstance>> get() {
-		return Flux.defer(() -> Flux.just(delegate.getInstances(serviceId)));
+		return serviceInstances.collectList().flatMapMany(Flux::just);
```

Comment:
Seems odd, but not sure I know any better. @smaldini or @simonbasle?

Reply:
*if I'm reading that right*
if the `serviceInstances` is cold, ie it reflects the instances at the time it is subscribed to, then it is legit to remove `Flux.defer`. The original use of `defer` was to avoid eagerly capturing the `List` immediately on assembly.

The `flatMapMany` is overkill if the intent is to convert a `Mono<List>` to a `Flux<List>` (same content): use `.flux()`.

Reply:
You're reading that right. Because we have a blocking and reactive service discovery now, we're keeping the reference of the flux. Both observables are cold (blocking via `Flux.defer`) so we're going to get the latest service instances on each subscribe.

Thanks for the `.flux()` hint! I'll adapt accordingly. 🙂